### PR TITLE
WIP | ✨ Major improvements

### PR DESCRIPTION
**Pull request by yedhrab**

### CODE REVIEW

File: ahk/templates/window/run.ahk [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{% extends "base.ahk" %}
+{% block body %}
+Run, {{ target }}, {{working_dir}}, {{options}}, output
+{% if wait %}
+WinWait, ahk_pid %output%
+WinGet, output, ID, ahk_pid %output%
```
Comment (spyoungtech) [inline ahk/templates/window/run.ahk]: If, for whatever reason this fails, do we need error handling around this?

File: ahk/window.py [hunk 2]
```diff
@@ -514,3 +518,51 @@ def find_windows_by_class(self, class_name, exact=False):
     def find_window_by_class(self, *args, **kwargs):
         with suppress(StopIteration):
             return next(self.find_windows_by_class(*args, **kwargs))
+
+    def run_window(self, target: str, working_dir="", options="") -> Window:
+        """Run command
```
Comment (spyoungtech) [inline ahk/window.py]: This is not a blocker to merging, but I would prefer if docstrings were in [sphinx docstring](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html) format.

Soon, I plan to write full documentation for readthedocs in rst and, as far as i know, autodoc only uses sphinxy docstrings :)

File: ahk/window.py [hunk 3]
```diff
@@ -514,3 +518,51 @@ def find_windows_by_class(self, class_name, exact=False):
...
+            command="Run",
+            target=target,
+            working_dir=working_dir,
+            options=options,
+            wait=True
+        )
+        ahk_id = self.run_script(script)
+        return self.find_window_by_id(ahk_id)
+
+    def run(self, target, working_dir="", options=""):
```
Comment (spyoungtech) [inline ahk/window.py]: Would it be possible for this to accept a `wait` keyword argument that would use `RunWait`? (should probably rename the internal name `wait`)?

Alternatively, maybe a `run_wait` function for RunWait would work, too.

File: tests/unittests/test_command.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+class TestCommand(TestCase):
+
+    def setUp(self):
+        self.ahk = AHK()
+
+    def test_run(self):
+        pid = self.ahk.run("notepad")
+        self.assertNotIsInstance(pid, Window)
+
+        time.sleep(0.5)
```
Comment (spyoungtech) [inline tests/unittests/test_command.py]: is this sleep in the right place?

File: tests/unittests/test_window.py [hunk 5]
```diff
@@ -71,13 +69,18 @@ def test_names(self):
...
+        self.win.close()
+        self.assertFalse(self.win.exist)
 
 
 if __name__ == "__main__":
+    import logging
+    import coloredlogs
+
+    coloredlogs.install(level='DEBUG')
+    logger = logging.getLogger(__name__)
```
Comment (spyoungtech) [inline tests/unittests/test_window.py]: This doesn't seem to be used anywhere.

**Comment by yedhrab** [comment]
I'm sorry about this newbie PR 🙄. I wiil improve it.

**Comment by spyoungtech** [comment]
> I'm sorry about this newbie PR 🙄. I wiil improve it.
> 
> > Additionally, How can I use Appveyor test without PR? Also, I open PR for these tests only 😅

Ah, no worries. If you put `WIP` in your PR title, I'll know it's still being worked on :)

As for AppVeyor, you can link your GitHub account to AppVeyor and add your ahk fork by clicking "+ New Project" on ci.appveyor.com -- It should pickup on the configuration from the appveryor.yml file automatically.

**Comment by yedhrab** [comment]
Sorry to late response, I'm busy these days. I'll do the things that you said soon

**Comment by spyoungtech** [comment]
Closing this for inactivity, but feel free to reopen any time.

Thanks again for the contributions

**Pull request closed without merging**